modules/light-clients/08-wasm/internal/types/store.go

```diff
@@ -0,0 +1,227 @@
+package types
+
+import (
+	"bytes"
+	"errors"
+	"io"
+
+	wasmvmtypes "github.com/CosmWasm/wasmvm/v2/types"
+
+	"cosmossdk.io/store/cachekv"
+	"cosmossdk.io/store/tracekv"
+	storetypes "cosmossdk.io/store/types"
+)
+
+var (
+	_ wasmvmtypes.KVStore = &StoreAdapter{}
+	_ storetypes.KVStore  = &MergedClientStore{}
+
+	SubjectPrefix    = []byte("subject/")
+	SubstitutePrefix = []byte("substitute/")
+)
+
+// MergedClientStore combines two KVStores into one.
+//
+// Both stores are used for reads, but only the subjectStore is used for writes. For all operations, the key
+// is checked to determine which types to use and must be prefixed with either "subject/" or "substitute/" accordingly.
+// If the key is not prefixed with either "subject/" or "substitute/", a default action is taken (e.g. no-op for Set/Delete).
+type MergedClientStore struct {
```

Comment:
Not sure about this name... Merging for me implies that two things are combined into one, but it's not really the case here, we are just grouping two things together... I actually don't dislike the name we had before...

Reply:
But two things are combined into one! 😛 

This is a struct with two `storetypes.KVStore` fields defined within it!

Reply:
The previous name gave a clear indication of where this type is used, while the new one is more generic. Since probably we will not use this anywhere else, I think it would make sense to be more explicit. If I think of merging, I would think of both stores merging into one, which is not the case. But I am fine, if you guys are happy with this naming.

Reply:
`ClientRecoveryStore` would probably get the ronseal approval award - does exactly what it says on the tin